src/modules/job-manager/plugins/age.c

```diff
@@ -0,0 +1,84 @@
+/************************************************************\
+ * Copyright 2020 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* builtins/age.c - basic priority plugin with age factor
+ *
+ */
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <flux/core.h>
+
+static double first_job_timestamp = -1.;
+
+static unsigned int age_factor (double t_submit)
+{
+    unsigned int inverse_age = (t_submit - first_job_timestamp)*10;
+    return (FLUX_JOB_PRIORITY_MAX - inverse_age);
+}
+
+static unsigned int urgency_factor (int urgency)
+{
+    double ppu = FLUX_JOB_PRIORITY_MAX/FLUX_JOB_URGENCY_MAX;;
+    return urgency * ppu;
+}
+
+static unsigned int age_based_priority (int urgency, double t_submit)
+{
+    if (first_job_timestamp  < 0.) {
+        first_job_timestamp = t_submit;
+    }
+    return (0.5 * age_factor (t_submit) + 0.5 * urgency_factor (urgency));
+}
+
+/*
+ *  Set a priority based on time in queue. (The first job submitted gets
+ *   the highest priority.
+ */
+static int priority_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *data)
+{
+    int urgency = 16;
+    double t_submit = 0.;
+    unsigned int priority;
+    flux_plugin_arg_unpack (args, FLUX_PLUGIN_ARG_IN,
```

Comment:
It seems like the practice of setting a default value arg value and then not checking the return value of arg_unpack() might make some errors like a misspelled argument name or incorrect type hard to track down.

Reply:
Yeah, return value checks are missing here.

Reply:
This has been fixed, the lack of return code checks was a holdover from early development and not meant ot be standard practice.